VocaDbWeb/webpack.mix.js

```diff
@@ -0,0 +1,194 @@
+const mix = require('laravel-mix');
+
+/*
+ |--------------------------------------------------------------------------
+ | Mix Asset Management
+ |--------------------------------------------------------------------------
+ |
+ | Mix provides a clean, fluent API for defining some Webpack build steps
+ | for your Laravel application. By default, we are compiling the Sass
+ | file for the application as well as bundling up all the JS files.
+ |
+ */
+
+require('laravel-mix-merge-manifest');
+
+mix
+	.mergeManifest()
+	.setPublicPath('./')
+	.webpackConfig({
+		output: {
+			library: 'app'
+		}
+	})
+
+
+	/*.scripts([
+		"Scripts/jquery-2.2.1.js",
+		"Scripts/bootstrap.js",
+		//"Scripts/jquery-ui-1.10.1.js", // doesn't work if bundled together
+		"Scripts/knockout-3.4.1.js",
+		"Scripts/knockout.punches.min.js",
+		"Scripts/lodash.js",
+		"Scripts/qTip/jquery.qtip.js",
+		"Scripts/marked.js"
+	], "bundles/shared/libs.js")
+
+	.scripts([
+		"Scripts/jquery-ui-1.10.4.js"
+	], "bundles/shared/jqui.js")*/
+	.js("Scripts/libs.js", "bundles/shared")
+
+	// SHARED BUNDLES
+	// Legacy common scripts - should be phased out
+	.scripts(["Scripts/VocaDB.js"], "bundles/VocaDB.js")
+
+	.ts("Scripts/App.ts", "bundles")
+
+	// Included on all pages (including front page)
+	// Generally the references go from viewmodels -> repositories -> models -> support classes
+	.scripts([
+	], "bundles/shared/common.js")
+
+	// Included on all pages except the front page (to optimize front page load time).
+	.scripts([
+		"Scripts/moment-with-locales.js",
+	], "bundles/shared/main.js")
+
+	// Included on all entry edit and create pages (album, artist, my settings etc.)
+	.scripts([
+		"Scripts/knockout-sortable.js"
+	], "bundles/shared/edit.js")
+
+	.scripts([
+		"Scripts/jquery.tools.min.js"	// REVIEW
+	], "bundles/Home/Index.js")
+
+	.scripts([
+		"Scripts/jqwidgets27/jqxcore.js", "Scripts/jqwidgets27/jqxrating.js"
+	], "bundles/jqxRating.js")
+
+
+	// VIEW-SPECIFIC BUNDLES
+	.scripts([
+	], "bundles/ActivityEntry/Index.js")
+
+
+	.scripts([
+	], "bundles/Album/Create.js")

```

Comment:
I'm not sure I understand the purpose of these scripts commands. Does this produce a file called Create.js? What's in that file?

Reply:
Yes, this produces an empty file called Create.js to prevent 404 errors. I didn't remove those scripts commands so that I can split App.ts into page-specific scripts later. In case it would be OK to get rid of the page-specific bundles and move all of application code to the shared bundles, we can remove them, or at least we should comment them (plus the corresponding `RenderScripts` in .cshtml files) out before merging this pull request. By the way, I use `App.ts` (and `App.js`) rather than `app.ts` so that I can easily rename the `app` variable later (e.g. to `vdb`).

Reply:
Ah okay. Yeah that's fine. Could you add a comment explaining this?

Reply:
I've added comments.